packages/cli/src/commands/hydrogen/deploy.ts

```diff
@@ -112,47 +138,98 @@ export default class Deploy extends Command {
 
 interface OxygenDeploymentOptions {
   environmentTag?: string;
+  force: boolean;
+  noJsonOutput: boolean;
   path: string;
   publicDeployment: boolean;
   shop: string;
   token?: string;
+  metadataDescription?: string;
   metadataUrl?: string;
   metadataUser?: string;
   metadataVersion?: string;
 }
 
 interface GitCommit {
   refs: string;
+  hash: string;
 }
 
 export async function oxygenDeploy(
   options: OxygenDeploymentOptions,
 ): Promise<void> {
   const {
     environmentTag,
+    force: forceOnUncommitedChanges,
+    noJsonOutput,
     path,
     shop,
     publicDeployment,
     metadataUrl,
     metadataUser,
     metadataVersion,
   } = options;
-  const ci = ciPlatform();
+  let {metadataDescription} = options;
+
+  let isCleanGit = true;
+  try {
+    await ensureIsClean(path);
+  } catch (error) {
+    if (error instanceof GitDirectoryNotCleanError) {
+      isCleanGit = false;
```

Comment:
Considering we are catching the error, is there any reason to use `ensureIsClean()` here instead of directly `isClean()` (returns true/false)?

Reply:
There's a theoretical possibility the project dir is a non-git directory (i.e. doesn't have the `.git` folder). Whilst recommended we don't require a repository directory.
In that scenario `isClean()` returns false and `ensureIsClean` returns a `GitError` (fatal: not a git repository) but we want to proceed, as telling the user to commit in a non-git directory doesn't make sense.

Reply:
We could, but the only function that cli-kit exports to that extent is [ensureInsideGitDirectory](https://github.com/Shopify/cli/blob/6ca39b787c124aa98e909d2a72b45ea1daa96487/packages/cli-kit/src/public/node/git.ts#L265C23-L265C47). Again that doesn't return a simple boolean, but a `OutsideGitDirectoryError` when outside of a Git folder, so I don't think that approach would be messier.